src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_ITEMS_NUMBER = 10;
+
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_ITEMS_NUMBER];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index < size && index >= 0) {
+            if (size == elementData.length) {
+                elementData = grow();
+            }
+            Object[] newElementData = new Object[elementData.length];
+            System.arraycopy(elementData, 0, newElementData, 0, index);
+            newElementData[index] = value;
+            System.arraycopy(elementData, index, newElementData, index + 1, size - index);
+            elementData = newElementData;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index Out of Bounds");
+        }
 
     }
 
+    private Object[] grow() {
+        Object[] newElementData = new Object[elementData.length + (elementData.length >> 1)];
+        System.arraycopy(elementData, 0, newElementData, 0, size);
+        return newElementData;
+    }
+
     @Override
     public void addAll(List<T> list) {
-
+        Object[] newElementData = new Object[size + list.size()];
+        System.arraycopy(elementData, 0, newElementData, 0, elementData.length);
+        elementData = newElementData;
```

Comment:
use `grow()` or create overloaded `grow(int sizeToGrow)`
```suggestion
        grow(parameters);
```